services/web/src/components/ConfirmModal.js

```diff
@@ -0,0 +1,77 @@
+import { useState, useCallback } from 'react';
+import { Button, Group, Stack } from '@mantine/core';
+import PropTypes from 'prop-types';
+import ErrorMessage from './ErrorMessage';
+import { modals } from '@mantine/modals';
+
+/**
+ * Confirm dialog component using Mantine Modal.
+ *
+ * @param {object} props
+ * @param {React.ReactNode} props.content - Content.
+ * @param {string} [props.confirmButton] - Confirm button label.
+ * @param {boolean} [props.negative] - If true, confirm button is red.
+ * @param {function} [props.onConfirm] - Async function called on confirm.
+ * @param {function} props.close - Function to close the modal.
+ * @returns {JSX.Element}
+ */
```

Comment:
does all this stuff need documentation? Does this provide us intellisense?

Reply:
yep it does, i was thinking why not. Specially i can generated it.